packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1041,6 +1043,23 @@ abstract class ParentDataWidget<T extends RenderObjectWidget> extends ProxyWidge
     return result;
   }
 
+  /// Write the data from this widget into the given render object's parent data.
+  ///
+  /// The framework calls this function whenever it detects that the
+  /// [RenderObject] associated with the [child] has outdated
+  /// [RenderObject.parentData]. For example, if the render object was recently
+  /// inserted into the render tree, the render object's parent data might not
+  /// match the data in this widget.
+  ///
+  /// Subclasses are expected to override this function to copy data from their
+  /// fields into the [RenderObject.parentData] field of the given render
+  /// object. The render object's parent is guartuneed to have been created by a
+  /// widget of type `T`, which usually means the render object's parent data
+  /// inherits from a given class.
+  ///
+  /// If this function modifies data that can change the parent's layout, this
+  /// function is responsible for calling [RenderObject.markNeedsLayout] on the
+  /// parent.
```

Comment:
i guess this could also affect only paint? should we mention that case that here?